File: Numerical_Integration/Monte-Carlo_Method/Monte-Carlo_Method.cpp
#include <stdlib.h>
#include <iostream>
#include <math.h>
#include <random>
#include <iomanip>
using namespace std;
#define f(x) sin(x) / (exp(x) + 1)

float MCM(double a, double b, int maxtrials)
{
    float integral = 0;

    for (int i = 0; i < maxtrials; i++)
    {
        random_device seed;
        uniform_real_distribution<float> distrib(a, b);
        double x0 = distrib(seed);

        integral += f(x0);
    }

    return integral * ((b - a) / (maxtrials - 1));
}

int main()
{
    int maxtrials;
    double a, b;

    cout << "Enter the upper limit:  ";
    cin >> b;
    cout << "Enter the lower limit:  ";
    cin >> a;
    cout << "Set number of trials : ";
    cin >> maxtrials;

    cout << "The solution of the integral is = " << MCM(a, b, maxtrials);

    return 0;
}

File: Numerical_Integration/Monte-Carlo_Method/Monte-Carlo_Method.py
from sympy import symbols, lambdify
import random

def monte_carlo_method(function, a, b, maxtrials):
    integral = 0.0
    for _ in range(maxtrials):
        integral += function(random.uniform(a,b))
    return integral*((b-a)/(maxtrials-1))

x = symbols('x')

input_expr = input("Enter the function f(x): ")
function = lambdify(x, input_expr, 'numpy')

a = float(input("Enter the lower limit a: "))
b = float(input("Enter the upper limit b: "))
maxtrials = int(input("Set maximum number of trials: "))

print("The solution of the integral is = ", monte_carlo_method(function, a, b, maxtrials))
